docs/source/developers/components_exports.rst

```diff
@@ -0,0 +1,486 @@
+The paths are always resolved relative to the package directory and must point
+to locations within it.
+
+When refering to objects (functions, classes, etc), Python name in dotted path
+format is used (e.g., ``hooks.on_initialize``). These names are resolved
+relative to the component's package (the package is prefixed to the name before
+importing). Multiple dots will be treated as a single dot, and any leading dots
+will be stripped. The names must point to importable objects. In most cases,
+failure to import an object will result in omission of that object.
+
+Options with mutiple values
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Some options allow one or more values. For a single value we write::
+
+    option = value
+
+For multiple values, we use a slightly different format::
+
+    option =
+        value 1
+        value 2
+
+Remember to add a newline before the first value and indent all lines until the
+next option. Each line is treated as a separate value.
+
+Databases
+---------
+
+Components can use one or more databases. Use of a database is completely
+optional, and components that do not specify any databases will not have them.
+
+There are two configuration options that are used for configuring databases::
+
+    [exports]
+
+    databases = mycomponent
+
+    migrations = migrations.package
+
+If a components wants to use a database, the first option is required. The
+``databases`` option can be used to specify one or more database that the
+component wants to use.
+
+The ``migrations`` option is used to point to a Python package that contains
+database migrations. This option is optional, and the default value is
+``migrations``. The package names are resolved relative to the component's
+package name. The package must contain subpackages that match the database
+name(s).
+
+For more information on working with databases, see
+:doc:`working_with_databases`.
+
+.. warning::
+    Try to pick a database name that will be reasonably unique across
+    components.
+
+Templates
+---------
+
+By default, templates used in a component are looked for in a directory called
+'views', located inside the component's package directory. This can be changed
+by specifying one or more alternative locations. For example::
+
+    [exports]
+
+    templats = templates
+
+It is important to remember that templates are resolved relative to template
+driectories, regardless of how many leves of hierarchy there is between the
+package directory and the template directory.
+
+To illustrate template resolution, let's take a look at a concrete example.
+Let's say the directory layout is as follows::
+
+    package/
+        templates/
+            foo/
+                foo1.tpl
+                foo2.tpl
+            bar/
+                baz/
+                    baz1.tpl
+                    baz2.tpl
+                bar1.tpl
+                bar2.tpl
+
+If we used the second configuration from the examples above, and we ask for a
```

Comment:
There is one example of config and one example of folder hierarchy under the Templates section, so it's unclear which is referring to.